website/static/js/apiPersonalToken.js

```diff
@@ -0,0 +1,421 @@
+/*
+ *  Knockout views for the Personal Access Token pages: List and Create/Edit pages.
+ */
+
+
+'use strict';
+
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var historyjs = require('exports?History!history');
+
+var ko = require('knockout');
+require('knockout.validation');
+var Raven = require('raven-js');
+
+var ChangeMessageMixin = require('js/changeMessage');
+var koHelpers = require('./koHelpers');  // URL validators etc
+var $osf = require('./osfHelpers');
+var oop = require('js/oop');
+var language = require('js/osfLanguage');
+var moment = require('moment');
+
+
+/*
+ *  Store the data related to a single API Personal Token
+ */
+var TokenData = oop.defclass({
+    constructor: function (data) {  // Read in API data and store as object
+        data = data || {};
+        var attributes = data.attributes || {};
+
+        // User-editable fields
+        this.name = ko.observable(attributes.name)
+            .extend({required: true, maxLength:200});
+
+        this.scopes = ko.observableArray(attributes.scopes ? attributes.scopes.split(' ') : undefined)
+            .extend({required: true});
+
+        this.token_id = ko.observable(attributes.token_id);
+
+        // Other fields. Owner and client ID should never change within this view.
+        this.id = data.id;
+        this.lastUsed = moment.utc(attributes.date_last_used).toString().split(' GMT')[0] + ' UTC';
+
+        this.owner = attributes.user_id;
+        this.webDetailUrl = data.links ? data.links.html : undefined;
+        this.apiDetailUrl = data.links ? data.links.self : undefined;
+
+        // Enable value validation in form
+        this.validated =  ko.validatedObservable(this);
+
+        this.isValid = ko.computed(function () {
+            return this.validated.isValid();
+        }.bind(this));
+    },
+
+    serialize: function () {
+        return { // Convert data to JSON-serializable format consistent with JSON API v1.0 spec
+            data: {
+                id: this.id,
+                type: 'tokens',
+                attributes: {
+                    name: this.name(),
+                    scopes: this.scopes().toString().replace(',', ' ') || [],
```

Comment:
By default, str.replace in JavaScript only replaces the first occurrence. The first argument should specify the global option.

```
[1,2,3].toString().replace(',', '')
"12,3"
```